test/system/generic/CMakeLists.txt

```diff
@@ -3,6 +3,9 @@ collect (PROJECT_LIB_TESTS alloc.c)
 collect (PROJECT_LIB_TESTS irq.c)
 collect (PROJECT_LIB_TESTS mutex.c)
 collect (PROJECT_LIB_TESTS atomic.c)
+collect (PROJECT_LIB_TESTS threads.c)
+
+collect (PROJECT_LIB_TEST_ALL threads.c)
```

Comment:
Seems not make sense to add threads test for generic as it is bare metal ( no OS).
could we remove it?